### PR TITLE
build: buildkit section

### PR DESCRIPTION
**Pull request by crazy-max**
~needs https://github.com/moby/buildkit/pull/3220~
needs https://github.com/docker/cli/pull/3832

Adds "BuildKit" section describing the BuildKit project the most basic way with cross linking to existing pages. Also adds a "Custom Dockerfile syntax" page reworked from Dockerfile reference documentation.

Following these changes I also merged the "build enhancements", "enabling buildkit" and "creating a dockerfile" pages in build section with proper redirects and cross linking.

Signed-off-by: CrazyMax <crazy-max@users.noreply.github.com>

### CODE REVIEW

File: build/extend/index.md
Comment (dvdksn) [inline build/extend/index.md:61]: for later: I think we will want to lift the description of "what's a BuildKit frontend" to a more prominent/visible place. I was confused about why it was called frontends until I read these paragraphs, and now it totally makes sense. Let's leave it here for now though!

**Review by dvdksn — COMMENTED** [review]
This addition, and especially the LLB description, is so damn good I am gonna print it, frame it, and put it on a wall. I have some rephrasing suggestions, that's all.

File: build/extend/dockerfile-frontend.md
Comment (jedevc) [inline build/extend/dockerfile-frontend.md:26]:
```suggestion
Dockerfile. The BuildKit backend allows seamlessly using external
```

File: build/extend/dockerfile-frontend.md
Comment (jedevc) [reply build/extend/dockerfile-frontend.md:84]: As another option, `experimental features`.

File: build/extend/index.md
Comment (crazy-max) [reply build/extend/index.md:30]: I think we should avoid talking about the "legacy" builder but the overall build experience. "skip redundant" is not entirely true, some are really optimized like merge/diff operations.

File: Dockerfile
Comment (crazy-max) [inline Dockerfile:64]: Temporarily ignore build_enhancements linking to internal hash new-docker-build-secret-information in htmlproofer while waiting for changes in Dockerfile reference on BuildKit repo:

```
#0 0.096 - ./_site/engine/reference/builder/index.html
#0 0.096   *  linking to internal hash #new-docker-build-secret-information that does not exist (line 2547)
#0 0.096      <a href="/develop/develop-images/build_enhancements/#new-docker-build-secret-information">[“build images with BuildKit”](https://github.com/develop/develop-images/build_enhancements/#new-docker-build-secret-information)</a>
```

Comment (crazy-max) [reply Dockerfile:64]: Opened https://github.com/moby/buildkit/pull/3220

**Review by jedevc — DISMISSED** [review]
LGTM once the corresponding buildkit PR is merged.

**Comment by crazy-max** [comment]
```
Ran on 1842 files!


- ./_site/engine/reference/commandline/build/index.html
  *  linking to internal hash #buildkit that does not exist (line 849)
     <a href="/engine/reference/builder/#buildkit">BuildKit backend</a>
  *  linking to internal hash #buildkit that does not exist (line 1114)
     <a href="/engine/reference/builder/#buildkit">enable BuildKit</a>
  *  linking to internal hash #buildkit that does not exist (line 1156)
     <a href="/engine/reference/builder/#buildkit">enable BuildKit</a>
```

Oh more links need to be changed

File: build/building/packaging.md
Comment (dvdksn) [inline build/building/packaging.md:33]: Remove parenthesis

```suggestion
The default filename to use for a Dockerfile is `Dockerfile`, without a file
extension. Using the default name allows you to run the `docker build` command
```

File: build/building/packaging.md
Comment (dvdksn) [inline build/building/packaging.md:99]: Remove italics

```suggestion
While optional, this directive instructs the Docker builder what syntax to use
```

File: language/python/build-images.md
Comment (dvdksn) [inline language/python/build-images.md:15]:
```suggestion
* You understand basic Docker concepts, and you have completed the orientation
  and setup in Get started [Part 1](../../get-started/index.md).
* You're familiar with the [Dockerfile format](../../build/building/packaging.md#dockerfile).
* You have [enabled BuildKit](../../build/buildkit/index.md#getting-started)
  on your machine.
```

File: language/nodejs/build-images.md
Comment (dvdksn) [inline language/nodejs/build-images.md:17]:
```suggestion
* You understand basic Docker concepts, and you have completed the orientation
  and setup in Get started [Part 1](../../get-started/index.md).
* You're familiar with the [Dockerfile format](../../build/building/packaging.md#dockerfile).
* You have [enabled BuildKit](../../build/buildkit/index.md#getting-started)
  on your machine.
```

File: language/java/build-images.md
Comment (dvdksn) [inline language/java/build-images.md:15]:
```suggestion
* You understand basic Docker concepts, and you have completed the orientation
  and setup in Get started [Part 1](../../get-started/index.md).
* You're familiar with the [Dockerfile format](../../build/building/packaging.md#dockerfile).
* You have [enabled BuildKit](../../build/buildkit/index.md#getting-started)
  on your machine.
```

File: language/golang/build-images.md
Comment (dvdksn) [inline language/golang/build-images.md:20]:
```suggestion
* You understand basic Docker concepts, and you have completed the orientation
  and setup in Get started [Part 1](../../get-started/index.md).
* You're familiar with the [Dockerfile format](../../build/building/packaging.md#dockerfile).
* You have [enabled BuildKit](../../build/buildkit/index.md#getting-started)
  on your machine.
```

**Review by dvdksn — COMMENTED** [review]
LGTM; bunch of nits

**Comment by crazy-max** [comment]
Thanks for the review @dvdksn and @jedevc. Will wait for https://github.com/docker/cli/pull/3832 to be merged to fix the htmlproofer issue.

**Review by dvdksn — APPROVED** [review]

**Pull request merged**
